File: Data Structures & Algorithms/find-pivot-index/submission-0.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        prefix=[]
        s=0
        for i in nums:
            s+=i
            prefix.append(s)
        suffix=[]
        nums=nums[::-1]
        s=0
        for i in nums:
            s+=i
            suffix.append(s)
        suffix=suffix[::-1]
        for i in range(len(prefix)):
            prefix[i]=suffix[i]-prefix[i]
        if 0 in prefix:
            return prefix.index(0)
        else:
            return -1

File: Data Structures & Algorithms/special-array-i/submission-0.cpp
class Solution {
public:
    bool isArraySpecial(vector<int>& nums) {
        if(nums.size()==1 || nums.size()==0)return true;
        int me=0,mo=0,oc=0,ec=0;
        for(int i=1;i<nums.size();i++){
            if((nums[i]%2)==(nums[i-1]%2))return false;
        }
        return true;
    }
};